File: workback/src/utils/responseHandler.js
class ResponseHandler {
  /**
   * 成功响应
   */
  success(res, data = null, message = 'Success', statusCode = 200) {
    return res.status(statusCode).json({
      success: true,
      message,
      data,
      timestamp: new Date().toISOString()
    });
  }

  /**
   * 错误响应
   */
  error(res, message = 'Error', statusCode = 500, errorCode = null) {
    return res.status(statusCode).json({
      success: false,
      message,
      errorCode,
      timestamp: new Date().toISOString()
    });
  }

  /**
   * 验证错误响应
   */
  validationError(res, errors, message = '请求参数验证失败') {
    return res.status(400).json({
      success: false,
      message,
      errors,
      timestamp: new Date().toISOString()
    });
  }

  /**
   * 未授权响应
   */
  unauthorized(res, message = '未授权访问') {
    return res.status(401).json({
      success: false,
      message,
      timestamp: new Date().toISOString()
    });
  }

  /**
   * 禁止访问响应
   */
  forbidden(res, message = '禁止访问') {
    return res.status(403).json({
      success: false,
      message,
      timestamp: new Date().toISOString()
    });
  }

  /**
   * 资源未找到响应
   */
  notFound(res, message = '资源未找到') {
    return res.status(404).json({
      success: false,
      message,
      timestamp: new Date().toISOString()
    });
  }
}

module.exports = {
  responseHandler: new ResponseHandler()
}; 

File: workbackpy/utils/response_handler.py
from flask import jsonify
from datetime import datetime

class ResponseHandler:
    """统一响应处理器"""
    
    @staticmethod
    def success(message="操作成功", data=None, status_code=200):
        """成功响应"""
        response = {
            "success": True,
            "message": message,
            "timestamp": datetime.now().isoformat(),
            "data": data
        }
        return jsonify(response), status_code
    
    @staticmethod
    def error(message="操作失败", status_code=400, error_code=None):
        """错误响应"""
        response = {
            "success": False,
            "message": message,
            "timestamp": datetime.now().isoformat(),
            "error_code": error_code
        }
        return jsonify(response), status_code
    
    @staticmethod
    def paginated_success(message="获取数据成功", data=None, pagination=None, status_code=200):
        """分页成功响应"""
        response = {
            "success": True,
            "message": message,
            "timestamp": datetime.now().isoformat(),
            "data": data,
            "pagination": pagination
        }
        return jsonify(response), status_code 